### PR TITLE
chore: Refactoring code to show plugin based icons for module instances on EE

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

Refactoring code to show plugin based icons for module instances on EE

Fixes [#30163](https://github.com/appsmithorg/appsmith/issues/30163)

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!IMPORTANT]  
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/8256707865>
> Commit: `897cc28f1c8d15efb7789788aba89687b85a4916`
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=8256707865&attempt=2" target="_blank">Click here!</a>
> All cypress tests have passed 🎉🎉🎉

## Summary by CodeRabbit

- **New Features**
	- Introduced a new interface for module metadata, including properties like module and plugin identifiers.
	- Added a new type property to the Module interface to distinguish module types.
	- Enhanced resource list loading with unique identifiers for each resource.
	- Implemented functionality to dynamically resolve and display icons based on module or plugin properties.
- **Enhancements**
	- Updated the applications page to support workspaces more effectively, including active workspace identification.
	- Improved module selection and display across the application, allowing for a more integrated module management experience.
- **Refactor**
	- Removed unused functions and components related to module icon resolution.
	- Streamlined the logic for fetching and displaying module-related information, including in action selectors and query editors.
- **Bug Fixes**
	- Fixed an issue with resource object identification by introducing unique `id` fields.
- **Style**
	- Added new class selector for module instance name editing, enhancing the styling options.

### CODE REVIEW

File: app/client/src/ce/selectors/modulesSelector.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import type { AppState } from "@appsmith/reducers";
+import type { Module } from "@appsmith/constants/ModuleConstants";
+
+export const getAllModules = (
+  state: AppState,
+): Record<string, Module> | any => {};
```
[the remark this answers is not in the document]
Comment (ankitakinger) [reply app/client/src/ce/selectors/modulesSelector.ts:7]: Intentional change

File: app/client/src/components/editorComponents/WidgetQueryGeneratorForm/CommonControls/DatasourceDropdown/useSource/useConnectToOptions.tsx [hunk 2]
```diff
@@ -97,13 +98,16 @@ interface ConnectToOptionsProps {
...
-        <Icon name="module" />
-      </EntityIcon>
-    );
+    const q = query as ModuleInstanceData;
+    const module = modules[q.config.sourceModuleId];
+    return resolveIcon({
+      iconLocation: pluginImages[module.pluginId] || "",
+      pluginType: module.pluginType,
+      moduleType: module.type,
+    });
```
[the remark this answers is not in the document]
Comment (ankitakinger) [reply app/client/src/components/editorComponents/WidgetQueryGeneratorForm/CommonControls/DatasourceDropdown/useSource/useConnectToOptions.tsx]: Done

File: app/client/src/ce/pages/Applications/index.tsx [hunk 3]
```diff
@@ -408,18 +407,14 @@ export function LeftPane(props: LeftPaneProps) {
...
-                workspace && (
-                  <WorkspaceMenuItem
-                    isFetchingWorkspaces={isFetchingWorkspaces}
-                    key={workspace?.id}
-                    selected={workspace?.id === activeWorkspaceId}
-                    workspace={workspace}
-                  />
-                ),
-            )}
+          {workspaces.map((workspace) => (
```
Comment (ashit-rath) [inline app/client/src/ce/pages/Applications/index.tsx:410]: Good job!

**Review by ashit-rath — APPROVED** [review]

**Pull request merged**